server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1644,6 +1644,7 @@ public static Setting<List<String>> stringListSetting(String key, List<String> d
         return new ListSetting<>(key, null, s -> defValue, Setting::parseableStringToList, v -> {}, properties) {
             @Override
             public List<String> get(Settings settings) {
+                checkDeprecation(settings);
```

Comment:
This was actually missing in this short-cut but was revealed by the monitoring tests so fixing it here.